pkg/collector/python/datadog_agent.go

```diff
@@ -609,3 +612,70 @@ func ObfuscateMongoDBString(cmd *C.char, errResult **C.char) *C.char {
 	// memory will be freed by caller
 	return TrackedCString(obfuscatedMongoDBString)
 }
+
+var (
+	telemetryMap = new(sync.Map)
+)
+
+func lazyInitTelemetryHistogram(checkName string, metricName string) telemetry.Histogram {
+	var key = checkName + "." + metricName
+	histogram, _ := telemetryMap.LoadOrStore(key, telemetryimpl.GetCompatComponent().NewHistogramWithOpts(
+		checkName,
+		metricName,
+		nil,
+		fmt.Sprintf("Histogram of %s for Python check %s", metricName, checkName),
+		[]float64{10, 25, 50, 75, 100, 250, 500, 1000, 10000},
+		telemetry.DefaultOptions,
+	))
+	return histogram.(telemetry.Histogram)
+}
+
+func lazyInitTelemetryCounter(checkName string, metricName string) telemetry.Counter {
+	var key = checkName + "." + metricName
+	counter, _ := telemetryMap.LoadOrStore(key, telemetryimpl.GetCompatComponent().NewCounterWithOpts(
+		checkName,
+		metricName,
+		nil,
+		fmt.Sprintf("Counter of %s for Python check %s", metricName, checkName),
+		telemetry.DefaultOptions,
+	))
+	return counter.(telemetry.Counter)
+}
+
+func lazyInitTelemetryGauge(checkName string, metricName string) telemetry.Gauge {
+	var key = checkName + "." + metricName
+	gauge, _ := telemetryMap.LoadOrStore(key, telemetryimpl.GetCompatComponent().NewGaugeWithOpts(
+		checkName,
+		metricName,
+		nil,
+		fmt.Sprintf("Gauge of %s for Python check %s", metricName, checkName),
+		telemetry.DefaultOptions,
+	))
+	return gauge.(telemetry.Gauge)
+}
+
+// EmitAgentTelemetry records a telemetry data point for a Python integration
+// NB: Cross-org agent telemetry needs to be enabled for each metric in
+// comp/core/agenttelemetry/impl/config.go defaultProfiles
+//
+//export EmitAgentTelemetry
+func EmitAgentTelemetry(checkName *C.char, metricName *C.char, metricValue C.float, metricType *C.char) {
+	goCheckName := C.GoString(checkName)
+	goMetricName := C.GoString(metricName)
+	goMetricValue := float64(metricValue)
+	goMetricType := C.GoString(metricType)
+
+	switch goMetricType {
+	case "counter":
+		counter := lazyInitTelemetryCounter(goCheckName, goMetricName)
+		counter.Add(goMetricValue)
+	case "histogram":
+		histogram := lazyInitTelemetryHistogram(goCheckName, goMetricName)
+		histogram.Observe(goMetricValue)
+	case "gauge":
+		gauge := lazyInitTelemetryGauge(goCheckName, goMetricName)
+		gauge.Add(goMetricValue)
+	default:
+		log.Warnf("EmitAgentTelemetry: unsupported metric type %s", goMetricType)
```

Comment:
Would it make sense to throw and exception when an invalid string is used?

Or at least include the check id in the error message, otherwise it would be very difficult to pinpoint the source of these warnings.